File: src/main/java/com/fastcampus/innercircle/survey_api/domain/entity/Response.java
package com.fastcampus.innercircle.survey_api.domain.entity;

import jakarta.persistence.*;

import java.time.LocalDateTime;
import java.util.ArrayList;
import java.util.List;

@Entity
public class Response {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long responseId;

    @Column(name = "form_id", nullable = false)
    private Long formId;

    @OneToMany(mappedBy = "response", cascade = CascadeType.ALL)
    private List<Answer> answers = new ArrayList<>();

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt = LocalDateTime.now();

    public Long getResponseId() {
        return responseId;
    }

    public List<Answer> getAnswers() {
        return answers;
    }

    public LocalDateTime getCreatedAt() {
        return createdAt;
    }

    public void setFormId(Long formId) {
        this.formId = formId;
    }

    public void setAnswers(List<Answer> answers) {
        this.answers = answers;
    }
}

File: src/main/kotlin/org/example/survey/domain/Response.kt
package org.example.survey.domain

import jakarta.persistence.CascadeType
import jakarta.persistence.Entity
import jakarta.persistence.FetchType
import jakarta.persistence.GeneratedValue
import jakarta.persistence.GenerationType
import jakarta.persistence.Id
import jakarta.persistence.JoinColumn
import jakarta.persistence.ManyToOne
import jakarta.persistence.OneToMany
import jakarta.persistence.Table
import java.time.LocalDateTime

@Entity
@Table(name = "response")
class Response(
    val respondedAt: LocalDateTime? = null,
    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "form_id")
    val form: Form,
    @OneToMany(mappedBy = "response", cascade = [CascadeType.ALL], fetch = FetchType.LAZY)
    val answers: List<Answer> = mutableListOf(),
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    val responseId: Long = 0,
)
